### ISSUE TITLE
Nginx config parser returns an error when config file ends with newline

### ISSUE THREAD

**Issue by roobre** [newrelic/nri-nginx#80]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description

Nginx config file parser fails with an error if the file ends with an empty newline.

## Expected Behavior

Nginx config file parser fails should handle this, as it is common and mandated by POSIX.

## Details

When a newline is found, subsequent spaces and newlines are consumed in a loop. Subsequently, `reader.UnreadRune()`. Is called. However, `UnreadRune()` will return an error if a rune was not read in the last operation (as it's the case for EOF).

https://github.com/newrelic/nri-nginx/blob/128c9be869a26cee3fc7f5b570bb80e9b0b4a2af/src/inventory.go#L60-L64

**Comment by roobre** [comment]
Reopening until we make a release

**Comment by stuart-c-moore** [comment]
Hi,

Before I possibly raise a new issue (or speak to new relic support), is this issue possibly related to this error I am seeing?


```
May 26 17:00:51 jira-LA newrelic-infra-service: time="2021-05-26T17:00:51Z" level=error msg="Integration command failed" error="exit status 1" instance=nginx-server-inventory integration=com.newrelic.nginx prefix=config/nginx stderr="[FATAL] can't continue: bufio: invalid use of UnreadRune\n" working-dir=/var/db/newrelic-infra/newrelic-integrations
```

The nginx.conf file itself is valid as far as nginx is converned (`nginx -t` passes)

**Comment by roobre** [comment]
@stuart-c-moore 
Yes it is! This is an ongoing incident right now, which should be fixed when we make a release. We are working actively on shipping this as soon as possibe, and should happen within this week.

Known workarounds are:
- Downgrade to the previous version of the integration
- Remove the trailing newline from the `nginx.conf` file

**Comment by roobre** [comment]
Fix released in v3.1.1 🎉
